Core/include/Acts/Geometry/detail/Layer.ipp

```diff
@@ -143,9 +143,15 @@ std::vector<SurfaceIntersection> Layer::compatibleSurfaces(
     if (!acceptSurface(sf, sensitive)) {
       return;
     }
+    bool boundaryCheck = options.boundaryCheck;
+    if (std::find(options.externalSurfaces.begin(),
+                  options.externalSurfaces.end(),
+                  sf.geometryId()) != options.externalSurfaces.end()) {
+      boundaryCheck = false;
+    }
```

Comment:
Do you think we'd need a `options.boundaryCheckNavigation` and one for `options.boundaryCheckExternal` ?
I am not sure.

Reply:
I don't think that would be needed. If you had 'options.boundaryCheckExternal=true' that be the same thing as not having external surface (you can just keep 'options.externalSurfaces={}'). I wouldn't cost anything to have an extra option but I don't think this is needed.